### PR TITLE
Wins refactor

### PR DESCRIPTION
**Pull request by jbubar**
## adds to issue #162

1. adds see more functionality for mobile and desktop view
2. Refactors the code to be more readable and modular.

screencapture-localhost-4000-wins-2020-11-15-13_51_46
screencapture-localhost-4000-wins-2020-11-15-13_52_45
screencapture-localhost-4000-wins-2020-11-15-13_52_31

problems yet to be sorted: The expanded does not show all of the text. I tried many different css properties for height, and checked the computed height and I could not figure it out. 
Screen Shot 2020-11-15 at 2 10 35 PM
Screen Shot 2020-11-15 at 2 10 42 PM

Still need to get the avatar photo working. @efrenmarin45 and I are on that.

Major thanks to @rblaz001 , for creating the modal overlay for desktop view. The code was really clean and taught me a lot.

### CODE REVIEW

**Review by drubgrubby — CHANGES_REQUESTED** [review]
@jbubar - This review is just in relation to the "...see more" functionality.  I'm going to spend a little more time with the refactoring of the code to make sure I understand it a little better before commenting. 

In the meantime... 

The "see more" feature has some issues, especially with mobile on Firefox and Safari, that should probably be addressed before merging. 

**Chrome**
Fullscreen in chrome works great.  When you click on "...see more" the overlay card pops up and when you click outside the card it goes away.  Very cool!

It does seem like if there is too much text it truncates it at some point. 

Is this supposed to show all of the text on one card?  Is the card supposed to scroll? (I feel like there was a discussion of this, but I don't remember if or how it was resolved). 

Screen Shot 2020-11-16 at 7 49 49 PM

On mobile, while the see more functionality does let you see more, it doesn't seem to display all of the text. 

Screen Shot 2020-11-16 at 7 30 51 PM
Screen Shot 2020-11-16 at 7 31 01 PM

Additionally, I'm guessing because of the absolute positioning of the see more div, the bottom third of the last line of text is cut off when "see less" is displayed. 

**Firefox**
In firefox the overlay also pops up with the correct information when you click "...see more" as it should.  However, it also truncates the text at a certain point. 

Firefox mobile, sadly, doesn't display more when you click "...see more".  In fact, it displays less.

Screen Shot 2020-11-16 at 7 54 37 PM
Screen Shot 2020-11-16 at 7 54 45 PM

**Safari**

In Safari, the last line of text is beheaded in the desktop view.  

Screen Shot 2020-11-16 at 7 58 01 PM

In safari mobile, the see more div is also partially covering up the text in an unfortunate way, and all of the text doesn't display when you click "...see more".  

Screen Shot 2020-11-16 at 8 10 04 PM
Screen Shot 2020-11-16 at 8 10 13 PM

In summary, it seems like there are three main issues that should be addressed: 
1) In mobile views, the "see more" does not display all of the text.  
2) The see more div sometimes partially covers unintended text. 
3) The full text isn't displayed when you click "...see more". (As I mentioned above, there may have been a solution to this discussed that I'm just blanking on.  If so, please remind me).

**Comment by jbubar** [comment]
@drubgrubby I made the changes you requested. thanks for your comprehensive review

**Review by drubgrubby — APPROVED** [review]

**Comment by drubgrubby** [comment]
@jbubar made all of the changes from my previous review, which was primarily a visual/functional review of the "see more..." functionality.  Good work!

I have looked at the refactoring of the code, and it looks good to me.  Not being the most experienced JavaScript coder in creation (yet), I can't be 100% sure that there's nothing there that I've missed, but it looks good to me and I give it my okay to merge. 

Two notes: 
1) I think it would be good to remove "James Bond" from the list of cards.
2) The "see more..." option still shows even when the content isn't very long, and there isn't any more to see. If you remove James Bond then this won't be an immediate issue, and I'd recommend it for a future issue.  But in a perfect world, we should only see "see more..." when there is more to see.

**Comment by ExperimentsInHonesty** [comment]
@jbubar please add a conditional statement that only shows the see more when there is content over the length at which text is hidden.

FYI @drubgrubby

**Comment by jbubar** [comment]
Updated the PR to not include see more if not needed
screencapture-localhost-4000-wins-2020-11-29-11_02_43
screencapture-localhost-4000-wins-2020-11-29-11_03_07

**Review by drubgrubby — APPROVED** [review]
Changes approved.  "see more" now only shows when there is actually more to see.  @jbubar very clever way that you did this.  Nice work!
